students/john_rogers/lesson04/mailroom_pt2.py

```diff
@@ -0,0 +1,183 @@
+#!/usr/bin/env python3
+"""
+mailroom.py -
+    1) prompt user for 3 actions - Send thank you, create report or quit
+Author: JohnR
+Version: 1.3
+Last updated: 12/29/2018
+Notes: TODO: create a single template using the .format(**d) method
+
+  v2.0 requirements:
+    1) use dict where appropriate
+    2) create a dict-based menu
+    3) create a single template using .format(**d) method for the letter
+    4) write a full set of letters to everyone on disk in individual files
+    5) add a function that goes through all donors, generates a thank you
+        letter, and writes it to disk as a text file.
+        A) use donor name and date for file name
+"""
+
+
+def main():
+    """
+    Create a list of donor and amounts, then get user input for the
+    various actions available.
+    :return: none
+    """
+
+    # make some global variables to experiment with
+    global db
+    global main_prompt
+    global main_dispatch
+
+    # create a main database
+    db = {'sting': {'d1': 13.45,
+                    'd2': 214.34,
+                    'd3': 453.23,
+                    },
+          'bono': {'d1': 54.54,
+                   'd2': 778.01,
+                   'd3': 564.35,
+                   },
+          'oprah': {'d1': 66.34,
+                    'd2': 664.33,
+                    'd3': 566.45,
+                    },
+          'yoko': {'d1': 64.24,
+                   'd2': 67.03,
+                   'd3': 990.34,
+                   },
+          'santa': {'d1': 45.43,
+                    'd2': 98345.32,
+                    'd3': 54.34,
+                    'd4': 456.23,
+                    },
+         }
+
+    # create the main user prompt
+    main_prompt = (
+        "\nWelcome to the main menu!\n"
+        "Please pick a number from the following:\n"
+        "1: exit the program\n"
+        "2: check donor list and become a donor\n"
+        "3: create an interactive report of current donors\n"
+        "4: write a thank you note for each donor and write to file\n"
+        ">>>\n"
+    )
+
+    # create the main menu using dict switch
+    main_dispatch = {
+        '1': exit_menu,
+        '2': thank_you,
+        '3': create_report,
+        '4': write_report,
+    }
+
+    # calling menu with no variables at this point since they are global
+    menu()
+
+
+def menu():
+    """
+    Get user input
+    :return: call the appropriate menu item
+    """
+    while True:
+        response = input(main_prompt)
+        if response == '1':
+            exit_menu()
```

Comment:
why special if statement? you have "1" mapped to this function in `main_dispatch`